src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts

```diff
@@ -1,99 +1,304 @@
+	readonly advancedExport = ExportMethodEnum.ADVANCED;
+	readonly pdfFormat = FormatEnum.PDF;
+
+	title = 'Export';
+	description = 'keep in mind that the image may be protected';
+	exportMethod: ExportMethodEnum = ExportMethodEnum.BASIC;
+	pdfExportMapId: string;
+	exporting: boolean;
+	graphicExport = [GraphicExportEnum.All, GraphicExportEnum.DrawsAndMeasures, GraphicExportEnum.North, GraphicExportEnum.Description];
+	graphicexportMap = new Map(this.graphicExport.reduce((entries: Array<[string, boolean]>, e) => {
+		entries.push([e, true]);
+		return entries;
+	}, []));
+	formats = [FormatEnum.JPG, FormatEnum.PDF];
+	format = this.formats[0];
+	quality = DEFAULT_QUALITY;
 
 	@AutoSubscription
-	onMapReadyToExport$ = combineLatest(this.mapBlob, this.onClick).pipe(
-		tap(([blob, isExport]: [Blob, boolean]) => {
-			if (isExport) {
-				saveAs(blob, 'map.jpg');
-				this.logger.info(LOGS.success);
-			}
-			else {
-				this.logger.info(LOGS.canceled);
-			}
-			this.closeModal()
-		})
+	onActiveMapChange$ = this.store$.pipe(
+		select(selectActiveMapId),
+		tap( (activeMapId) => this.pdfExportMapId = activeMapId)
 	);
-
 	@AutoSubscription
-	isMinimalView$ = this.store$.select(selectIsMinimalistViewMode).pipe(
+	basicExport$ = this.store$.pipe(
+		select(selectIsMinimalistViewMode),
 		filter(Boolean),
-		tap( this.exportMapsToPng.bind(this))
+		debounceTime(100),
+		tap(this.basicExportMap.bind(this))
 	);
 
+	_pageSize = DEFAULT_PAGE_SIZE;
+
+	get pageSize() {
+		if (this.isPDF()) {
+			return this._pageSize;
+		}
+		return 'Screenshot';
+	}
+
+	set pageSize(value) {
+		this._pageSize = value
+	}
+
+	_pageSizes = { a0: [1189, 841], a1: [841, 594], a2: [594, 420], a3: [420, 297], a4: [297, 210], a5: [210, 148] };
+
+	get pageSizes() {
+		if (this.isPDF()) {
+			return Object.keys(this._pageSizes);
+		}
+		return ['Screenshot']
+	}
+
+	_qualities = { low: 72, normal: 150, high: 300 };
+
+	get qualities() {
+		return Object.keys(this._qualities);
+	}
+
 	get config() {
 		return this.toolsConfigData.exportMap;
 	}
 
-	constructor(
-		public store$: Store<any>,
-		public credentialsService: CredentialsService,
-		public dialogRef: MatDialogRef<ExportMapsPopupComponent>,
-		protected logger: LoggerService,
-		@Inject(DOCUMENT) protected document: any,
-		@Inject(toolsConfig) public toolsConfigData: IToolsConfig) {
+	constructor(protected store$: Store<any>,
+				protected logger: LoggerService,
+				protected dialogRef: MatDialogRef<ExportMapsPopupComponent>,
+				protected imageryCommunicatorService: ImageryCommunicatorService,
+				protected translateService: TranslateService,
+				@Inject(DOCUMENT) protected document: any,
+				@Inject(toolsConfig) public toolsConfigData: IToolsConfig) {
 		this.logger.info(LOGS.request);
-		this.store$.dispatch(new SetMinimalistViewModeAction(true));
+		this.translateService.get('direction', '').subscribe( (direction) => this.direction = direction === 'rtl' ? 'rtl' : 'ltr')
```

Comment:
is the translateService unsubscribes by default?

Reply:
yes.